frontend/glideinFrontendDowntimeLib.py

```diff
@@ -16,33 +16,97 @@
+    """
+
     def __init__(self, fname):
+        """
+        Args:
+            fname (str): Path to the downtime file.
+        """
         self.fname = fname
 
-    # if check_time==None, use current time
     def checkDowntime(self, check_time=None):
-        rtn = checkDowntime(self.fname, check_time)
-        return rtn
+        """Check if the specified time falls within a downtime period.
+
+        Args:
+            check_time (int, optional): The time to check, in UNIX timestamp format.
+                Defaults to the current time.
+
+        Returns:
+            bool: True if the specified time is within a downtime period, False otherwise.
+        """
+        return checkDowntime(self.fname, check_time)
 
-    # add a scheduled downtime
     def addPeriod(self, start_time, end_time, create_if_empty=True):
+        """Add a scheduled downtime period to the file.
+
+        Args:
+            start_time (int): Start time of the downtime, in UNIX timestamp format.
+            end_time (int): End time of the downtime, in UNIX timestamp format. Use `None`
+                for an indefinite downtime.
+            create_if_empty (bool): Whether to create the file if it doesn't exist.
+
+        Returns:
+            int: 0 if the downtime was added successfully.
+        """
         return addPeriod(self.fname, start_time, end_time, create_if_empty)
 
-    # start a downtime that we don't know when it will end    # if start_time==None, use current time
     def startDowntime(self, start_time=None, end_time=None, create_if_empty=True):
+        """Start an indefinite downtime.
+
+        Args:
+            start_time (int, optional): Start time of the downtime, in UNIX timestamp format.
+                Defaults to the current time.
+            end_time (int, optional): End time of the downtime, in UNIX timestamp format.
+                Use `None` for indefinite downtime.
+            create_if_empty (bool): Whether to create the file if it doesn't exist.
+
+        Returns:
+            int: 0 if the downtime was started successfully.
+        """
         if start_time is None:
             start_time = int(time.time())
         return self.addPeriod(start_time, end_time, create_if_empty)
 
-    # end a downtime (not a scheduled one)    # if end_time==None, use current time
     def endDowntime(self, end_time=None):
+        """End the current downtime (not a scheduled one).
+
+        Args:
+            end_time (int, optional): End time of the downtime, in UNIX timestamp format.
+                Defaults to the current time.
+
+        Returns:
+            int: The number of downtimes ended.
+        """
         return endDowntime(self.fname, end_time)
 
     def printDowntime(self, check_time=None):
+        """Print active downtime periods.
+
+        Args:
+            check_time (int, optional): The time to check against, in UNIX timestamp format.
+                Defaults to the current time.
+
+        Returns:
```

Comment:
Is this redundant? I haven't seen any occurrences of `Returns:` being described as None when the method does not return anything.

Reply:
The method has a return statement. I added an explanation.